### PR TITLE
Update a few stacks

### PR DESCRIPTION
**Pull request by huseyin**
Modified stacks:
- `apache`
- `docker`
- `memsql`
- `nodejs`

### CODE REVIEW

File: apache.yml [hunk 1]
```diff
@@ -12,46 +12,27 @@ resource:
...
-        cat - <<EOF >> /home/$KODING_USER/.bashrc
-        echo "'=============================================================================='"
-        echo "To start apache server type: sudo /etc/apache2/bin/apachectl -k start"
-        echo "To stop apache server type: sudo /etc/apache2/bin/apachectl -k stop"
-        echo "'=============================================================================='"
+        KODING_USER=${var.koding_user_username}
+        DEFAULT_SHELL=$(basename $SHELL)
+        SHELLPROFILE=/home/$KODING_USER/."$DEFAULT_SHELL"rc
+
+        cat <<-EOF >>$SHELLPROFILE
```
Comment (szkl) [inline apache.yml:28]: Can we benefit from /etc/issue file instead of a shell-specific profile script?

Comment (szkl) [reply apache.yml:28]: s/issue/mot.d/

File: docker.yml [hunk 2]
```diff
@@ -14,16 +14,25 @@ resource:
...
+          linux-image-extra-virtual \
+          linux-image-extra-$(uname -r) \
+          --no-install-recommends
+        
+        curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+        add-apt-repository \
+          "deb https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+        
+        apt-get update
+        apt-get install -y docker-ce --no-install-recommends
```
Comment (szkl) [inline docker.yml:38]: Put named options before positional arguments.

File: apache.yml [hunk 3]
```diff
@@ -12,46 +12,27 @@ resource:
       tags:
         Name: '${var.koding_user_username}-${var.koding_group_slug}'
       user_data: |-
-        set -e
+        #!/bin/bash
+
+        set -xe
```
Comment (szkl) [inline apache.yml:17]: Long option names, please.

File: apache.yml [hunk 4]
```diff
@@ -12,46 +12,27 @@ resource:
...
+
         export DEBIAN_FRONTEND=noninteractive
         
-        #Getting username IP, and version variables
-        export KODING_USER=${var.koding_user_username}
-        IPVAR=$(curl http://169.254.169.254/latest/meta-data/public-ipv4)
-        version="2.4.25"
+        apt-get update
+        apt-get install -y apache2 --no-install-recommends
```
Comment (szkl) [inline apache.yml:22]: Put named options before positional arguments and use long option name.

File: docker.yml [hunk 5]
```diff
@@ -14,16 +14,25 @@ resource:
       tags:
         Name: '${var.koding_user_username}-${var.koding_group_slug}'
       user_data: |-
+        #!/bin/bash
+        
+        set -eo pipefail
+        set -x
```
Comment (szkl) [inline docker.yml:20]: Long option names.

File: docker.yml [hunk 6]
```diff
@@ -14,16 +14,25 @@ resource:
...
-        apt-get update -y
-        #Install Docker engine
-        sudo apt-get install apt-transport-https ca-certificates
-        sudo apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
-        echo 'deb https://apt.dockerproject.org/repo ubuntu-trusty main' > /etc/apt/sources.list.d/docker.list
-        sudo apt-get update
-        sudo apt-get purge -y lxc-docker
-        sudo apt-get install -y linux-image-extra-$(uname -r)
-        sudo apt-get install -y --force-yes docker-engine
+        apt-get update && apt-get install -y \
```
Comment (szkl) [inline docker.yml:24]: And operator is redundant because `errexit` will cause an early exit if a failure occurs.

File: docker.yml [hunk 7]
```diff
@@ -14,16 +14,25 @@ resource:
...
+          software-properties-common \
+          apt-transport-https \
+          curl \
+          ca-certificates \
+          linux-image-extra-virtual \
+          linux-image-extra-$(uname -r) \
+          --no-install-recommends
```
Comment (szkl) [inline docker.yml:31]: Should be placed before positional arguments.

File: docker.yml [hunk 8]
```diff
@@ -14,16 +14,25 @@ resource:
...
+        
+        curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
```
Comment (szkl) [inline docker.yml:33]: Long option names for readability's sake.

File: memsql.yml [hunk 9]
```diff
@@ -1,24 +1,37 @@
...
       instance_type: m4.2xlarge
       ami: ''
       tags:
         Name: '${var.koding_user_username}-${var.koding_group_slug}'
       user_data: |-
-        # :
+        #!/bin/bash
+        
+        set -euo pipefail
+        set -x
```
Comment (szkl) [inline memsql.yml:18]: Long option names.

File: memsql.yml [hunk 10]
```diff
@@ -1,24 +1,37 @@
...
-        #Single memsql installation
-        wget http://download.memsql.com/memsql-ops-5.1.0/memsql-ops-5.1.0.tar.gz
-        tar -xzvf memsql-ops-5.1.0.tar.gz
-        cd memsql-ops-5.1.0
-        yes| ./install.sh
+        
+        apt-get update
+        apt-get install -y wget --no-install-recommends
+
+        MEMSQL_VERSION="5.7.1"
```
Comment (szkl) [inline memsql.yml:25]: I tend to use camel case for variables which are not exported to subprocesses.

File: memsql.yml [hunk 11]
```diff
@@ -1,24 +1,37 @@
...
+        MEMSQL_NAME="memsql-ops-$MEMSQL_VERSION"
+        MEMSQL_TARBALL="$MEMSQL_NAME.tar.gz"
+
+        wget -qO- \
```
Comment (szkl) [inline memsql.yml:29]: Long option names.

File: memsql.yml [hunk 12]
```diff
@@ -1,24 +1,37 @@
...
+          "http://download.memsql.com/$MEMSQL_NAME/$MEMSQL_TARBALL" | tar -xvz
```
Comment (szkl) [inline memsql.yml:30]: Long option names for `tar` too.

File: nodejs.yml [hunk 13]
```diff
@@ -1,34 +1,37 @@
...
-      # we will tag the instance here so you can identify it when you login to your AWS console
       tags:
         Name: '${var.koding_user_username}-${var.koding_group_slug}'
-      # on user_data section we will write bash and configure our VM
       user_data: |-
-        set -e
+        #!/bin/bash
+
+        set -euo pipefail
+        set -x
```
Comment (szkl) [inline nodejs.yml:18]: Long option names.

File: nodejs.yml [hunk 14]
```diff
@@ -1,34 +1,37 @@
...
+
         export DEBIAN_FRONTEND=noninteractive
-        DISTRO=$(lsb_release -is | tr '[:upper:]' '[:lower:]')
-        CODENAME=$(lsb_release -cs)
-        VERSION="node_6.x"
         
-        curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add -
+        apt-get update
+        apt install -y wget --no-install-recommends
```
Comment (szkl) [inline nodejs.yml:23]: Named argument should be placed before positional arguments.

File: nodejs.yml [hunk 15]
```diff
@@ -1,34 +1,37 @@
...
         
-        echo "deb https://deb.nodesource.com/$VERSION $CODENAME main" > /etc/apt/sources.list.d/nodesource.list
-        echo "deb-src https://deb.nodesource.com/$VERSION $CODENAME main" >> /etc/apt/sources.list.d/nodesource.list
+        NODE_VERSION="v6.10.2"
```
Comment (szkl) [inline nodejs.yml:25]: Can you make it fetch latest LTS version automatically?

File: nodejs.yml [hunk 16]
```diff
@@ -1,34 +1,37 @@
...
+        NODE_NAME="node-$NODE_VERSION-linux-x64"
+        NODE_TARBALL="$NODE_NAME.tar.xz"
+        NODE_DIST_URL="https://nodejs.org/dist/$NODE_VERSION/$NODE_TARBALL"
+
+        wget -qO- $NODE_DIST_URL | tar --extract --xz --strip-components=1 \
```
Comment (szkl) [inline nodejs.yml:30]: Long option names for `wget`.

**Review by szkl — CHANGES_REQUESTED** [review]
Thanks, @htaslan. It looks okay overall. I have requested some syntactical changes.

File: apache.yml (hunk 3 above)
Comment (huseyin) [reply apache.yml:17]: i disagree with you. `set -euo pipefail` is a typical use.  therefore i think that there is no need to use the long option names. but, if Koding is using it in this way, there is no problem. I can fix it.

Comment (szkl) [reply apache.yml:17]: I'm okay with it. I just don't make any exceptions for option names in shell scripts :)

**Pull request merged**